Sources/Bookmarks/BookmarkListViewModel.swift

```diff
@@ -138,6 +140,30 @@ public class BookmarkListViewModel: BookmarkListInteracting, ObservableObject {
         refresh()
     }
 
+    private func reattachOrphanedBookmarks(forMoving bookmark: BookmarkEntity, toIndex: Int) {
+        guard let rootFolder = BookmarkUtils.fetchRootFolder(context) else {
+            return
+        }
+
+        let orphanedBookmarks = bookmarks.filter { $0.parent == nil }
+        guard !orphanedBookmarks.isEmpty else {
+            return
+        }
+
+        let orphanedBookmarksToAttachToRootFolder: [BookmarkEntity] = {
+            let toIndexInOrphanedBookmarks = toIndex - rootFolder.childrenArray.count
+            guard bookmark.parent == nil else {
+                return toIndexInOrphanedBookmarks >= 0 ? Array(orphanedBookmarks.prefix(through: toIndexInOrphanedBookmarks)) : []
+            }
+            guard let bookmarkIndexInOrphans = orphanedBookmarks.firstIndex(where: { $0.uuid == bookmark.uuid }) else {
+                return [bookmark]
```

Comment:
If the bookmarks is not in orphans, we do ned to reattach it?

Reply:
Ah I see but we fallen through the parent check so the bookmark itself is orphan, this guard should never happen.

Reply:
Exactly, especially now that we're not refetching orphans, but just taking all without a parent. Can as well return an empty array here (and perhaps an assertion failure), but the expectation is that this doesn't happen.